app/models/infra_conversion_job.rb

```diff
@@ -45,13 +43,82 @@ def load_transitions
     }
   end
 
+  # Example state:
+  #   :state_name => {
+  #     :description => 'State description',
+  #     :weight      => 30,
+  #     :max_retries => 960
+  #   }
+  def load_states
+    {
+      :running => {
+        :max_retries => 5760 # 24 hours with a retry interval of 15 seconds
+      },
+      :post_conversion => {
+        :max_retries => 1680 # 7 hours with a retry interval of 15 seconds
+      }
+    }
+  end
+
+  def states
+    @states ||= load_states
+  end
+
   def migration_task
     @migration_task ||= target_entity
     # valid states: %w(migrated pending finished active queued)
   end
 
+  def on_entry(state_hash, _)
+    state_hash ||= { :state => 'active', :status => 'Ok', :message => states[state.to_sym].description }
+    if context["retries_#{state}".to_sym].to_i.zero?
+      state_hash[:started_on] = Time.now.utc
+      state_hash[:percent] = 0.0
+    end
+    state_hash
+  end
+
+  def on_retry(state_hash, state_progress = nil)
+    if state_progress.nil?
+      state_hash[:percent] = context["retries#{state}".to_sym].to_f / states[state.to_sym][:max_retries].to_f * 100.0
```

Comment:
I think `retries#{state}` is meant to be `retries_#{state}` here (missing underscore).